pkg/apis/messaging/v1/subscription_validation.go

```diff
@@ -57,6 +58,10 @@ func (ss *SubscriptionSpec) Validate(ctx context.Context) *apis.FieldError {
 	}
 
 	if !missingSubscriber {
+		ctx := ctx
+		if feature.FromContext(ctx).IsEnabled(feature.KReferenceGroup) {
```

Comment:
We should get ahead of this and make some kind of global registration that enables the mutation of flags in a smaller footprint, like 

```go
ctx := feature.FeatureContext(ctx)
```

and have a way to link the test for `IsEnabled(feature.KReferenceGroup)` with `duckv1.KReferenceGroupAllowed` automatically on the  `FeatureContext` call.

Reply:
TBH I don't get what you mean, It seems to me what you're talking about is what I already implemented here... `FromContext` already returns the `Flags` data structure, to check if a flag is enabled or not. No need for a global here...

Reply:
Ok I think I get what you mean, check out this: https://github.com/knative/eventing/pull/5440/commits/c677412acc0b3759d22eb559689ee2643dd48ce5